app/tests/src/wallet/account-manager.js

```diff
@@ -45,62 +40,40 @@ const initAccountManager = async (settings, i) => {
     config,
   })
 
-  accountManagers.push(
-    AccountManager({config, cryptoProvider, blockchainExplorer, maxAccountIndex})
-  )
+  return AccountManager({config, cryptoProvider, blockchainExplorer, maxAccountIndex})
 }
 
-before(async () => {
-  for (const accountManagerSetting of accountManagerSettings) {
-    await initAccountManager(accountManagerSetting)
-  }
+before(() => {
+  Object.entries(accountManagerSettings).forEach(([name, setting]) => {
+    accountManagers[name] = initAccountManager(setting)
+  })
 })
 
-describe('account discovery', () => {
-  // TODO: refactor to foreach
-  it('should discover the right amount of accounts', async () => {
-    const mockNet = mockNetwork(ADALITE_CONFIG)
-    mockNet.mockBulkAddressSummaryEndpoint()
-    const accounts = await accountManagers[0].discoverAccounts()
-    // console.log(accounts, accountManagerSettings[0])
-    assert.equal(accounts.length, accountManagerSettings[0].numberOfDiscoveredAccounts)
-    mockNet.clean()
-  })
-  it('should not discover accounts if bulk export is disabled', async () => {
-    const mockNet = mockNetwork(ADALITE_CONFIG)
-    mockNet.mockBulkAddressSummaryEndpoint()
-    const accounts = await accountManagers[1].discoverAccounts()
-    // console.log(accounts, accountManagerSettings[1])
-    assert.equal(accounts.length, accountManagerSettings[1].numberOfDiscoveredAccounts)
-    mockNet.clean()
-  })
-  it('should not discover accounts if wallet is not shelley compatible', async () => {
```

Comment:
sorry, why are these two getting removed?

Reply:
They havent really got removed, I am iterating over the settings, and that way I don't have to define a test for each one of them.
![image](https://user-images.githubusercontent.com/22474126/106457477-5428e800-648f-11eb-9c22-83f8bebc0832.png)